Gems/Atom/RHI/Metal/Code/Source/RHI.Builders/ShaderPlatformInterface.cpp

```diff
@@ -424,7 +424,7 @@ namespace AZ
             AZStd::string outMetalLibFile = RHI::BuildFileNameWithExtension(shaderSourceFile, tempFolder, "metallib");
             
             //Debug symbols are always enabled at the moment. Need to turn them off for optimized shader assets. 
-            AZStd::string shaderDebugInfo = "-gline-tables-only -MO";
+            AZStd::string shaderDebugInfo = "-gline-tables-only -MO -fpreserve-invariance";
```

Comment:
Will the -fpreserve-invariance be turned off when debug symbols is turned off for optimization in the future?

Reply:
Probably not

Reply:
Updated the PR with the -fpreserve-invariance moved to a separate msl to air options string.